scripts/generateDeepExports.mjs

```diff
@@ -0,0 +1,362 @@
+  const exportFile = path.join(exportDir, 'index.ts');
+
+  // Create directory if it doesn't exist
+  if (!existsSync(exportDir)) {
+    await fs.mkdir(exportDir, { recursive: true });
+  }
+
+  if (utilities.includes(componentName)) {
+    // Handle utility with re-export blocks
+    let existingContent = '';
+    if (existsSync(exportFile)) {
+      existingContent = await fs.readFile(exportFile, 'utf8');
+    }
+
+    const bestSourcePackage = await getBestSourcePackage(
+      targetPackagePath,
+      componentName,
+      sourcePackageChain,
+    );
+
+    // Check if already exported from the best source (outside of auto-generated blocks)
+    if (await isUtilityAlreadyExported(targetPackagePath, componentName, bestSourcePackage)) {
+      return;
+    }
+
+    // Remove existing block and add new one
+    let updatedContent = removeExistingBlock(existingContent);
+    updatedContent = addReExportBlock(updatedContent, bestSourcePackage, componentName);
+
+    // Only write if content changed
+    if (updatedContent !== existingContent) {
+      await fs.writeFile(exportFile, updatedContent);
+      console.warn(
+        `  ✓ Added re-export block for utility ${componentName} from @mui/${bestSourcePackage} in ${path.relative(process.cwd(), targetPackagePath)}`,
+      );
+    }
+    return;
+  }
+
+  // For regular components, export from the base package
+  const exportContent = `${IDENTIFIER_LINE}
+export * from '@mui/${sourcePackageName}/${componentName}';
+`;
+  await fs.writeFile(exportFile, exportContent);
+  console.warn(
+    `✓ Created deep export for ${componentName} in ${path.relative(process.cwd(), targetPackagePath)}`,
+  );
+}
+
+/**
+ * Process a base package and create deep exports for its components in target packages.
+ * @param {string} basePackageName
+ */
+async function processPackage(basePackageName) {
+  const basePackagePath = path.join(process.cwd(), 'packages', basePackageName);
+  /** @type {{targets: string[]; ignore: string[]; utilities: string[];}} */
+  const targetPackages = PACKAGE_MAPPINGS[basePackageName] || {};
+
+  if (!existsSync(basePackagePath)) {
+    console.warn(`Base package not found: ${basePackagePath}`);
+    return;
+  }
+
+  console.warn(`\n📦 Processing ${basePackageName} -> ${targetPackages.targets.join(' -> ')}...`);
+
+  // Get component directories from the base package
+  const componentDirectories = [
+    await getComponentDirectories(basePackagePath, targetPackages.ignore),
+  ];
+  const sourcePackageChain = [basePackageName];
+
+  for (const targetPackage of targetPackages.targets) {
+    const targetPackagePath = path.join(process.cwd(), 'packages', targetPackage);
+
+    if (!existsSync(targetPackagePath)) {
+      console.warn(`Target package not found: ${targetPackagePath}`);
+      return;
+    }
+
+    // Get additional components specific to pro/premium
+    // eslint-disable-next-line no-await-in-loop
```

Comment:
Should we follow this lint? It'd probably make the script slightly faster by processing files concurrently. 

Also not a blocker, as it doesn't affect the functionality of the script.

Reply:
No, these need to run in order

Reply:
Ah, you mean because the premium package depends on the pro? 

Yeah, makes sense. We could have some concurrency here, but probably not worth the extra complexity.